packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -194,36 +203,55 @@ class Hero extends StatefulWidget {
   final bool transitionOnUserGestures;
 
   // Returns a map of all of the heroes in context, indexed by hero tag.
-  static Map<Object, _HeroState> _allHeroesFor(BuildContext context, bool isUserGestureTransition) {
+  static Map<Object, _HeroState> _allHeroesFor(
+      BuildContext context,
+      bool isUserGestureTransition,
+      NavigatorState triggeringNavigator,
```

Comment:
This needs private docs. Also triggering seems a bit tight. Maybe transitioning?